swap/src/database.rs

```diff
@@ -3,13 +3,39 @@ use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use std::{fmt::Display, path::Path};
 use uuid::Uuid;
 
-mod alice;
-mod bob;
+pub mod alice;
+pub mod bob;
 
-pub use alice::*;
-pub use bob::*;
+pub use alice::Alice;
+pub use bob::Bob;
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub enum Swap {
+    Alice(Alice),
+    Bob(Bob),
+}
+
+impl From<Alice> for Swap {
+    fn from(from: Alice) -> Self {
+        Swap::Alice(from)
+    }
+}
+
+impl From<Bob> for Swap {
+    fn from(from: Bob) -> Self {
+        Swap::Bob(from)
+    }
+}
+
+impl Display for Swap {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Swap::Alice(alice) => Display::fmt(alice, f),
+            Swap::Bob(bob) => Display::fmt(bob, f),
+        }
+    }
+}
 
-#[derive(Debug)]
```

Comment:
Hum, did you *voluntarily* drop some clippy lints in the process?

Reply:
Fixed this in a follow up commit, not sure what happened here tbh